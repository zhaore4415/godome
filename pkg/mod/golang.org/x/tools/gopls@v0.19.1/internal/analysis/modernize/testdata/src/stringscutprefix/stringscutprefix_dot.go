package stringscutprefix

import (
	. "strings"
)

// test supported cases of pattern 1
func _() {
	if HasPrefix(s, pre) { // want "HasPrefix \\+ TrimPrefix can be simplified to CutPrefix"
		a := TrimPrefix(s, pre)
		_ = a
	}
}

// test supported cases of pattern2
func _() {
	if after := TrimPrefix(s, pre); after != s { // want "TrimPrefix can be simplified to CutPrefix"
		println(after)
	}
	if after := TrimPrefix(s, pre); s != after { // want "TrimPrefix can be simplified to CutPrefix"
		println(after)
	}
}
