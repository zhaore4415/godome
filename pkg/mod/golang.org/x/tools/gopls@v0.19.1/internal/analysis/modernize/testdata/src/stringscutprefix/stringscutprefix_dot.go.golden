package stringscutprefix

import (
	. "strings"
)

// test supported cases of pattern 1
func _() {
	if after, ok := CutPrefix(s, pre); ok { // want "HasPrefix \\+ TrimPrefix can be simplified to CutPrefix"
		a := after
		_ = a
	}
}

// test supported cases of pattern2
func _() {
	if after, ok := CutPrefix(s, pre); ok { // want "TrimPrefix can be simplified to CutPrefix"
		println(after)
	}
	if after, ok := CutPrefix(s, pre); ok { // want "TrimPrefix can be simplified to CutPrefix"
		println(after)
	}
}